src/cell.cpp

```diff
@@ -635,6 +635,143 @@ void DAGCell::to_hdf5(hid_t group_id) const { return; }
 
 #endif
 
+//==============================================================================
+// UniversePartitioner implementation
+//==============================================================================
+
+UniversePartitioner::UniversePartitioner(const Universe& univ)
+{
+  // Find all of the z-planes in this universe.  Add them to the surfs_ member.
+  // Use surf_set for O(1) searches that ensure surfs_ entries are not repeated.
+  std::unordered_set<int32_t> surf_set;
+  for (auto i_cell : univ.cells_) {
+    for (auto token : model::cells[i_cell]->rpn_) {
+      if (token < OP_UNION) {
+        auto i_surf = std::abs(token) - 1;
+        const auto* surf = model::surfaces[i_surf].get();
+        if (const auto* zplane = dynamic_cast<const SurfaceZPlane*>(surf)) {
+          if (surf_set.find(i_surf) == surf_set.end()) {
+            surf_set.insert(i_surf);
+            surfs_.push_back(i_surf);
+          }
+        }
+      }
+    }
+  }
+
+  // Define a lambda for comparing z-planes by their location, and use it to
+  // sort the surfs_ member.
+  std::sort(surfs_.begin(), surfs_.end(),
```

Comment:
Thinking about this a little more, it would be much easier to just use `std::set` instead of `std::unordered_set` above and then just construct `surfs_` by iterating over the `set`.